src/test/regress/sql/propagate_set_commands.sql

```diff
@@ -23,6 +23,53 @@ SET TRANSACTION READ ONLY;
 INSERT INTO test VALUES (2,2);
 END;
 
+BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;
+-- should reflect isolation level of current transaction
+SELECT current_setting('transaction_isolation') FROM test WHERE id = 1;
+END;
+
+START TRANSACTION ISOLATION LEVEL REPEATABLE READ;
+-- should reflect isolation level of current transaction
+SELECT current_setting('transaction_isolation') FROM test WHERE id = 1;
+END;
+
+BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE;
+-- should reflect isolation level of current transaction
+SELECT current_setting('transaction_isolation') FROM test WHERE id = 1;
+END;
+
+BEGIN READ ONLY;
+-- should reflect read-only status of current transaction
+SELECT current_setting('transaction_read_only') FROM test WHERE id = 1;
+END;
+
+BEGIN READ WRITE;
+-- should reflect read-only status of current transaction
+SELECT current_setting('transaction_read_only') FROM test WHERE id = 1;
+SET TRANSACTION READ ONLY;
+-- should reflect writable status of current transaction
+SELECT current_setting('transaction_read_only') FROM test WHERE id = 1;
+END;
+
+BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE DEFERRABLE;
+-- should reflect deferrable status of the current transaction
+SELECT current_setting('transaction_deferrable') FROM test WHERE id = 1;
+END;
+
+BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE DEFERRABLE READ ONLY;
+SELECT current_setting('transaction_read_only') FROM test WHERE id = 1;
+SELECT current_setting('transaction_deferrable') FROM test WHERE id = 1;
+END;
+
+SET default_transaction_isolation TO 'repeatable read';
+
+BEGIN;
```

Comment:
could be interesting to have tests without tx block and m-shard query?

SELECT DISTINCT current_setting('transaction_isolation') FROM test;